conda/core/solve.py

```diff
@@ -731,17 +732,16 @@ def _add_specs(self, ssc):
         # the only things we should consider freezing are things that don't conflict with the new
         #    specs being added.
         explicit_pool = ssc.r._get_package_pool(self.specs_to_add)
+        if ssc.prune:
+            # Ignore installed specs on prune.
+            installed_specs = ()
+        else:
+            installed_specs = tuple(concatv(
+                (_.to_match_spec() for _ in ssc.prefix_data.iter_records())
+            ))
```

Comment:
```suggestion
            installed_specs = tuple(
                *(_.to_match_spec() for _ in ssc.prefix_data.iter_records())
            )
```

Reply:
not quite, i think you need to use [`itertools.chain.from_iterable`](https://docs.python.org/3/library/itertools.html#itertools.chain.from_iterable) instead